src/index.ts

```diff
@@ -133,6 +160,10 @@ export class Err<E> implements BaseResult<never, E> {
     mapErr<E2>(mapper: (err: E) => E2): Err<E2> {
         return new Err(mapper(this.val));
     }
+
+    flatMap(mapper: (val: never) => Result<never, E>): Result<never, E> {
```

Comment:
There should be a corresponding `flatMap` declaration in `BaseResult` with a doc comment:
```typescript
    flatMap<T2, E2>(mapper: (val: T) => Result<T2, E2>): Result<T2, E | E2>;
```

Reply:
This could definitely be there for the documentation purposes but in terms of functionality I couldn't get it to play nice with all the other types.

Had tried many variations that you can find in a previous commit, but if you can work your magic around it and make it work that would be amazing!